Makefile

```diff
@@ -38,10 +41,14 @@ endif
 
 test:
 	# testing..
+	@$(GOFAIL_ENABLE)
 	CGO_ENABLED=1 go test -race -cover $(TEST_PKGS)
+	@$(GOFAIL_DISABLE)
 
 basic_test:
+	@$(GOFAIL_ENABLE)
 	go test $(BASIC_TEST_PKGS)
+	@$(GOFAIL_DISABLE)
```

Comment:
`go test $(BASIC_TEST_PKGS)` need be updated to `go test $(BASIC_TEST_PKGS) || $(GOFAIL_DISABLE)` so when the test failed, generated files will be restored.